docs/ui/prepare-migration/dataflow-tuning.md

```diff
@@ -0,0 +1,88 @@
+---
+layout: default
+title: Tune Dataflow (Optional)
+parent: Prepare Migration Page
+grand_parent: SMT UI
+nav_order: 4
+---
+
+# Tune Dataflow (Optional)
+{: .no_toc }
+
+In case of minimal downtime migration, the dataflow jobs launched by the Spanner Migration Tool can be **optionally** tuned with custom runtime environment variables such as MaxWorkers, NumWorkers, specifying [networks and subnetworks](https://cloud.google.com/dataflow/docs/guides/specifying-networks) etc. 
+
+
+{: .highlight }
+
+To configure dataflow, first specify the target database in the 'Configure Spanner Database' step. This enables the configure button for the remaining steps.
+
+![](https://services.google.com/fh/gumdrop/preview/misc/dataflow-form.png)
+
+<details open markdown="block">
+  <summary>
+    Table of contents
+  </summary>
+  {: .text-delta }
+1. TOC
+{:toc}
+</details>
+
+{: .highlight }
+
+Currently, we only allow configuring a subset of options available in Dataflow via SMT. For the full list of options available, click [here](https://cloud.google.com/dataflow/docs/reference/pipeline-options).
+
+### VPC Host ProjectId
+Specify the project id of the VPC that you want to use. This is required in order to use private connectivity. By default, this is assumed to be the same as Spanner project. Ensure this is specified if also specifying a network and subnetwork.
+
+If using a shared VPC, a common practice is to have it in a separate project. Ensure this field specifies the correct host project for shared VPC use cases.
+
+Present under the Networking section of the form. 
+
+### VPC Network
+Specify the name of the VPC network to use. For private connectivity, specify both the VPC network and subnetwork. If no network and subnetwork is provided, the [default](https://cloud.google.com/dataflow/docs/guides/specifying-networks#specifying_a_network_and_a_subnetwork) network is used.
+
+Present under the Networking section of the form. 
+
+### VPC Subnetwork
+Specify the name of the VPC subnetwork to use. For private connectivity, specify both the VPC network and subnetwork. If no network and subnetwork is provided, the [default](https://cloud.google.com/dataflow/docs/guides/specifying-networks#specifying_a_network_and_a_subnetwork) network is used.
+
+Present under the Networking section of the form. 
```

Comment:
Let's add a note here about what we do under which type of configuration as far as enabling/disabling public IPs for the Dataflow VMs are concerned.

For example - We disable public IPs when running inside a VPC. On that note, I think we should also expose enabling/disabling public IPs directly on the UI. Krishna faced this issue and I think its a valid configuration to add.

Reply:
Added a note explaining this. 

As for the option to configure this, we decided to remove this based on recommendations for the dataflow team.